### ISSUE TITLE
[BUG] Las imágenes de algunos problemas históricos ya no se ven

### ISSUE THREAD

**Issue by darkmx** [omegaup/omegaup#6091]
Hola amigos de omegaUp.

En problemas viejitos (principalmente de Karel) como https://omegaup.com/arena/problem/Camina, ya no se ven las imágenes que estaban adjuntas. Aunque actualmente las imágenes se guardan en el markdown, en el pasado omegaUp las hospedaba en la carpeta /img (ej. https://omegaup.com/img/7a8bfc353f35dba36d0bd16b7a749311fe0423f3.png). Posiblemente con la migración murieron.

![image](https://user-images.githubusercontent.com/19719969/144749823-b7e0150d-fa0c-4cb3-8537-c5c8da5c8249.png)

**Comment by kukogit** [comment]
Help!!! Hoy tengo examen con mis chamacos!!!

**Comment by lhchavez** [comment]
cuáles son los problemas afectados?

**Comment by lhchavez** [comment]
parece que lo que hay que hacer es editar el statement para que tenga el nombre de la imagen correcto. por ejemplo, en Camina,

```markdown

![Entrada](/img/7a8bfc353f35dba36d0bd16b7a749311fe0423f3.png)

![Entrada](/img/aaca5b9125ae95c896d234b8e52a36cb548bb2a3.png)
```

hay que cambiar eso a

```markdown
# Ejemplo
![Entrada](in.png)

![Entrada](out.png)
```
```

**Comment by kukogit** [comment]
Los de estos concursos: 
https://omegaup.com/arena/ORIG2122PEI/
https://omegaup.com/arena/ORIG2122PM/

Algunos si muestran imágenes pero otros no

**Comment by darkmx** [comment]
intentaré modificar todos los problemas afectados en masa ahorita (con un script que sólo tengo que modificar), pero sería bueno si pudieran modificar esos a mano (para no programar tan a las prisas)

**Comment by kukogit** [comment]
El cambio sería solo ponerle in.png y out.png, y que hago en caso de que sean más imágenes??

**Comment by lhchavez** [comment]
no, depende de cómo se llamen las imágenes en el .zip. si son más imágenes se usa el nombre que tenga la imagen en el .zip

**Comment by kukogit** [comment]
Todos mis ejercicios tienen la imagen en el markdown. Pero creo que después de un tiempo, cuando bajaba los ZIPs de los problemas, ya aparecía una ruta de una imagen, pero yo generalmente siempre ponía la imagen en el markdown.

**Comment by lhchavez** [comment]
oh ya entonces sólo es cosa de editar el markdown para que sea el original con los nombres de la imagen en el markdown.

esas reescrituras de imagenes las dejamos de hacer hace tiempo, por eso solo afecta a los problemas subidos hace mucho.

**Comment by darkmx** [comment]
Veo esto: `{"statement":{"language":"es","images":{"in.png":"\/img\/Camina\/766819304a697fb748cb1c82127560e590b155a3.png","out.png":"\/img\/Camina\/4d1c74e3b33da43ffd97ab3c0eb28170f1ef8ea7.png"},"markdown":"# Descripci\u00f3n\n\nKarel quiere visitar a su abuelita, la casa de la abuelita se encuentra siempre adelante de karel, en la primera posici\u00f3n que tiene la izquierda y se esta junto a un zumbador.\n\n# Problema\nHaz un programa que lleve a Karel a la casa de su abuelita\n\n# Entrada\n\nKarel siempre inicia al principio del camino y orientado en direcci\u00f3n hacia la casa de su abuelita, el camino es recto y karel siempre encontrar\u00e1 a su abulita antes de caminar 30 pasos.\n\n# Salida\n\nSolo se evaluar\u00e1 la posicion final en que quede karel. que debe ser la primera posicion que cumpla con lo siguiente izquierda bloqueda y junto a un zumbador.\n\n# Ejemplo\n![Entrada](\/img\/7a8bfc353f35dba36d0bd16b7a749311fe0423f3.png)\n\n![Entrada](\/img\/aaca5b9125ae95c896d234b8e52a36cb548bb2a3.png)\n"},"settings":{"cases":[],"limits":{"ExtraWallTime":"0s","MemoryLimit":268435456,"OutputLimit":10240,"OverallWallTimeLimit":"1m0s","TimeLimit":"5s"},"validator":{"name":"token-caseless"}},"title":"Camina","alias":"Camina","input_limit":10240,"visits":117,"submissions":3132,"accepted":755,"difficulty":0.621857073441698,"creation_date":1381571772,"source":"Olimpiada Morelense de Rob\u00f3tica e Inform\u00e1tica","order":"normal","visibility":1,"email_clarifications":false,"quality_seal":false,"version":"f58c7c75cb6a413f1d864ec66af409291dbea05d","commit":"6b8387346723e3318117bb75dd25249ebe46ab46","problemsetter":{"username":"isailandao","name":"Isa\u00ed Landa","creation_date":1381571772},"languages":["kp","kj"],"points":10.46,"score":0,"exists":true,"status":"ok","_id":"5e8e478709d9b886745320"}`

Para confirmar: hay que poner el key en el markdown en lugar de la ruta, ¿cierto?

**Comment by lhchavez** [comment]
correcto, hay que reemplazar `/img/${basename(value)}` por `key` (porque ese mapa trae el nombre del problema embebido)

**Comment by kukogit** [comment]
Ya la mayoría se ve excepto: https://omegaup.com/arena/ORIG2122PM/#problems/K-Levantando-Totems

**Comment by kukogit** [comment]
Ya se ve... hace rato no se veía

> Ya la mayoría se ve excepto: https://omegaup.com/arena/ORIG2122PM/#problems/K-Levantando-Totems

Ya se ve... hace rato no se veía.

**Comment by darkmx** [comment]
OK, terminaré el script para modificar en masa los demás, pero lo ejecuto hasta la noche (ya que esté 100% seguro de que no tiene bugs)

**Comment by darkmx** [comment]
hay algunos casos bien raros Dx ¿ideas? 

![image](https://user-images.githubusercontent.com/19719969/144752009-0caf0ec4-bd1a-466d-a43e-8fe4556cd891.png)

**Comment by kukogit** [comment]
Gracias por el apoyo, ya luego veré lo del resto de los problemas. Los problemas que vaya creando no tendrán este problema, cierto?

**Comment by lhchavez** [comment]
correcto, cualquier problema que se suba a partir de hoy ya no va a tener este problema.

**Comment by kukogit** [comment]
Gracias.

**Comment by kukogit** [comment]
Y gracias por la ayuda express para el examen de mis chamacos!!!

**Comment by kukogit** [comment]
Dónde puedo la documentación de los cambios que hicieron, es decir, para saber si hay alguna función o herramienta nueva en OmegaUp?

**Comment by lhchavez** [comment]
la lista canónica de cambios está aquí https://github.com/omegaup/omegaup/commits/release

**Comment by lhchavez** [comment]
hay 1789 versiones de problemas afectadas. ETA para el fix automatizado: hoy en la noche.

**Comment by lhchavez** [comment]
ya se empezó a correr el fix automatizado y ya terminó con ~100 problemas. en teoría mañana termina de correr.

**Comment by lhchavez** [comment]
ya se terminó de correr el fix. Faltan 100 problemas que hay que arreglar semi-manualmente.

**Comment by lhchavez** [comment]
Ya sólo 17 problemas quedan por arreglar. Esos sí se van a tener que hacer manualmente

**Comment by darkmx** [comment]
¿Esos casos cuáles son? si tuvieras las imágenes (en un backup de la carpeta) yo lo arreglo

**Comment by lhchavez** [comment]
es más bien que se tiene que arreglar commit-por-commit porque se cambiaron las imágenes en varios commits y no basta con arreglar el último.

**Comment by lhchavez** [comment]
Listo, ya todos los problemas _deberían_ tener imágenes!
